homeassistant/components/surepetcare/__init__.py

```diff
@@ -83,40 +79,53 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         _LOGGER.error("Unable to connect to surepetcare.io: Wrong %s!", error)
         return False
 
-    spc = SurePetcareAPI(hass, surepy)
-    hass.data[DOMAIN][SPC] = spc
-
-    await spc.async_update()
+    async def _update_method() -> dict[int, SurepyEntity]:
+        """Get the latest data from Sure Petcare."""
+        try:
+            return await surepy.get_entities(refresh=True)
+        except SurePetcareError as err:
+            raise UpdateFailed(f"Unable to fetch data: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_method=_update_method,
+        update_interval=SCAN_INTERVAL,
+    )
 
-    async_track_time_interval(hass, spc.async_update, SCAN_INTERVAL)
+    hass.data[DOMAIN] = coordinator
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
This integration doesn't use config entries so we shouldn't use this api. Call refresh manually.